modules/cluser-logging-troubleshooting-loki-entry-out-of-order-errors.adoc

```diff
@@ -0,0 +1,124 @@
+:_module-type: PROCEDURE
+
+[id="cluser-logging-troubleshooting-loki-entry-out-of-order-messages_{context}"]
+= Troubleshooting Loki "entry out of order" errors
+
+Forwarding a large block of messages to a Grafana Loki logging system exceeds the rate limit and generates "entry out of order" errors. To fix this issue, you update some values in the Loki server configuration file, `loki.yaml`.
+
+.Prerequisites
+
+* You have configured the `ClusterLogForwarder` custom resource to forward logs to Grafana Loki.
+
+* Your system sends a block of messages that is larger than 2 MB to Loki, such as:
++
+----
+"values":[["1630410392689800468","{\"kind\":\"Event\",\"apiVersion\":\
+.......
+......
+......
+......
+\"received_at\":\"2021-08-31T11:46:32.800278+00:00\",\"version\":\"1.7.4 1.6.0\"}},\"@timestamp\":\"2021-08-31T11:46:32.799692+00:00\",\"viaq_index_name\":\"audit-write\",\"viaq_msg_id\":\"MzFjYjJkZjItNjY0MC00YWU4LWIwMTEtNGNmM2E5ZmViMGU4\",\"log_type\":\"audit\"}"]]}]}
+----
+
+* When you enter `oc logs -c fluentd`, the Fluentd logs in your OpenShift Logging cluster show the following messages:
++
+[source,text]
+----
+429 Too Many Requests Ingestion rate limit exceeded (limit: 8388608 bytes/sec) while attempting to ingest '2140' lines totaling '3285284' bytes
+
+429 Too Many Requests Ingestion rate limit exceeded' or '500 Internal Server Error rpc error: code = ResourceExhausted desc = grpc: received message larger than max (5277702 vs. 4194304)'
+----
+
+* When you open the logs on the Loki server, they display `entry out of order` messages like these:
++
+[source,text]
+----
+,\nentry with timestamp 2021-08-18 05:58:55.061936 +0000 UTC ignored, reason: 'entry out of order' for stream:
+
+{fluentd_thread=\"flush_thread_0\", log_type=\"audit\"},\nentry with timestamp 2021-08-18 06:01:18.290229 +0000 UTC ignored, reason: 'entry out of order' for stream: {fluentd_thread="flush_thread_0", log_type="audit"}
+----
+
+.Procedure
+
+* Update the following fields in the Loki server configuration file, `loki.yaml`, with the values shown here:
```

Comment:
Clarify this is part of the Loki server configuration (wherever that is), not the OCP cluster logging configuration.

Reply:
Updated. Thanks.